sys/sys/types.h

```diff
@@ -251,6 +251,12 @@ struct cap_rights;
 typedef	struct cap_rights	cap_rights_t;
 #endif
 
+#ifdef __CHERI_PURE_CAPABILITY__
```

Comment:
Is vm_offset_t (and thus potentially vm_pointer_t) used in userspace (which doesn't make a whole lot of sense IMO...), or should this be guarded by _KERNEL?

Reply:
Never mind I see it just below...

Reply:
`vm_offset_t` is used in user space in a few places that deal with virtual addresses as well as in boot loader code (stand/).  We may indeed end up wanting `vm_pointer_t` in the boot loader, but I'm fine with having this be under #ifdef _KERNEL for now and expanding that in the future to add || defined(STANDALONE) if we need it for a purecap boot loader.